File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    for char in s:
        if char == '(':
            stack.append(char)
        else:
            if not stack:
                answer = False
                break
            stack.pop()
    
    if stack:
        answer = False
    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        String[] strs = s.split("");
        
        Deque<String> dq = new ArrayDeque<>();
        for (String ss : strs) {
            if (ss.equals("(")) {
                dq.offer(ss);
            } else {
                if (dq.isEmpty()) {
                    answer = false;
                    break;
                }
                
                dq.pollLast();
            }
        }
        
        if (!dq.isEmpty()) {
            answer = false;
        }
        
        return answer;
    }
}